jujugui/static/gui/src/app/components/login/test-login.js

```diff
@@ -105,6 +106,63 @@ describe('LoginComponent', function() {
     assert.compareJSX(wrapper, expected);
   });
 
+  it('renders without the USSO login button', function() {
+    const wrapper = renderComponent({bakeryEnabled: false});
+    var expected = (
+      <div className="login">
+        <div className="login__logo">
+          <SvgIcon height="30" name="juju-logo" width="75" />
+        </div>
+        <div className="login__full-form">
+          <div className="login__env-name">
+            Login
+          </div>
+          {undefined}
+          <form
+            className="login__form"
+            onSubmit={wrapper.find('form').prop('onSubmit')}
+            ref="form">
+            <label
+              className="login__label">
+              Username
+              <input
+                className="login__input"
+                name="username"
+                ref="username"
+                type="text" />
+            </label>
+            <label
+              className="login__label">
+              Password
+              <input
+                className="login__input"
+                name="password"
+                ref="password"
+                type="password" />
+            </label>
+            <GenericButton
+              submit={true}
+              type="positive">
+              Login
+            </GenericButton>
+          </form>
+        </div>
+        <div className="login__message">
+          <p>
+            Find your username and password with<br />
+            <code>juju show-controller --show-password</code>
+          </p>
+          <div className="login__message-link">
+            <a href="https://jujucharms.com" target="_blank">
+              jujucharms.com
+            </a>
+          </div>
+        </div>
+      </div>
+    );
+    assert.compareJSX(wrapper, expected);
```

Comment:
Instead of comparing the whole output you can just check that the button is not there. e.g.

`assert.equal(wrapper.find('USSOLoginLink').length, 0);`

Reply:
That's a great idea, done.